### PR TITLE
feat: Create guild and brew when guild loggedIn

### LINKED ISSUE #25: Create collections mongodb
### Review our collections:

- races

### Create new collection
- guildCollection
- homebrewCollection

### Functions 
- Save guild when loggedIn
- Create default homeBrew for that Guild

### PR DESCRIPTION
**Pull request by math97**
This PR implements a creation of guild on our database when a new Guild loggedIn.Also create a homebrew for that guild.
resolves #25

### CODE REVIEW

File: src/controllers/registerGuildController.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+export const registerGuildController = async (guild: Guild): Promise<void> => {
+  const guildRepository = PrismaGuildRepository.getInstance()
+  const homeBrewRepository = PrismaHomebrewRepository.getInstance()
+
+  const registerNewGuildUseCase = new RegisterNewGuildUseCase(
+    guildRepository,
+    homeBrewRepository,
+  )
+  const description = guild.description ? guild.description : undefined
```
Comment (caio-emidio) [inline src/controllers/registerGuildController.ts]: unnecessary.
`const description = guild.description ? guild.description : undefined` is basically 
`const description = guild.description`

Any case if you want to do this type of description

const description = guild.description ?? undefined

File: src/controllers/registerGuildController.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  const guildToRegister = { guildId: guild.id, name: guild.name, description }
+
+  const { guild: guildRegistered, homebrew } =
+    await registerNewGuildUseCase.execute(guildToRegister, {
+      name: `${guild.name} Homebrews`,
+      description: `Homebrews for ${guild.name}`,
+    })
+
+  console.log('Guild registered:', guildRegistered)
```
Comment (caio-emidio) [inline src/controllers/registerGuildController.ts]: are all those `console.log` necessaries? Maybe we could implement any type of discord.log("") and all this information go directly to discord by "discord.reply"

File: src/repositories/prisma/prismaGuildRepository.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { Guild, Prisma } from '@prisma/client'
+import { prisma } from '../../lib/prisma'
+import { GuildRepository } from '../guildRepository'
+
+export class PrismaGuildRepository implements GuildRepository {
+  // eslint-disable-next-line no-use-before-define
+  private static instance: PrismaGuildRepository
```
Comment (caio-emidio) [inline src/repositories/prisma/prismaGuildRepository.ts]: if its possible, could we dont put here anything so Generic? 

`private static guildRepository: PrismaGuildRepository`
or maybe
`prismaGuildRepository`

in my perspective is better then instance.

File: src/repositories/prisma/prismaGuildRepository.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  static getInstance() {
+    if (this.instance) {
+      return this.instance
+    }
+    this.instance = new PrismaGuildRepository()
+    return this.instance
+  }
+
+  async create(data: Prisma.GuildCreateInput) {
+    const guild = await prisma.guild.create({
```
Comment (caio-emidio) [inline src/repositories/prisma/prismaGuildRepository.ts:18]: return `prisma.guid.create({ ... `

File: src/controllers/registerGuildController.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { Guild } from 'discord.js'
+import { PrismaGuildRepository } from '@/repositories/prisma/prismaGuildRepository'
+import { RegisterNewGuildUseCase } from '@/useCases/registerNewGuildUseCase'
+import { PrismaHomebrewRepository } from '@/repositories/prisma/prismaHomebrewRepository'
+
+export const registerGuildController = async (guild: Guild): Promise<void> => {
```
Comment (caio-emidio) [inline src/controllers/registerGuildController.ts]: We have to discuss whether we are going to keep OOP or functional programming. And if we keep both, in which case we use each of them.

File: src/repositories/prisma/prismaGuildRepository.ts (hunk 4 above)
Comment (math97) [reply src/repositories/prisma/prismaGuildRepository.ts:18]: the spread will generate conflict as prisma.guild.create really expect an object as data

**Pull request merged**